kore/src/Kore/Attribute/Pattern.hs

```diff
@@ -115,13 +132,24 @@ instance HasConstructorLike (Pattern variable) where
 
 simplifiedAttribute :: Pattern variable -> Simplified
 simplifiedAttribute Pattern {constructorLike, simplified}
-    | isConstructorLike constructorLike = Simplified
-    | otherwise = simplified
-
-isSimplified :: Pattern variable -> Bool
-isSimplified Pattern {constructorLike, simplified}
-  | isConstructorLike constructorLike = True
-  | otherwise = isFullySimplified simplified
+  | Simplified.isFullySimplified simplified = simplified
+  | isConstructorLike constructorLike =
+    error "Inconsistent attributes, constructorLike implies fully simplified."
+  | otherwise = simplified
```

Comment:
since this returns simplified whenever it does not error, it seems more like the combination of an assert and a getter.